src/utils.jl

```diff
@@ -0,0 +1,56 @@
+"""
+    safe_div(x, y)
+
+Safely divide `x` by `y`. If `y` is zero, return `x` directly.
+"""
+safe_div(x, y) = ifelse(iszero(y), x, x/y)
+
+"""
+    least_dims(idxs)
+
+Compute the least dimensions, of which array can be accessed by the indices `idxs`.
+"""
+least_dims(idxs::AbstractArray{<:Integer}) = (maximum(idxs), )
+
+function least_dims(idxs::AbstractArray{<:Tuple})
+    Tuple(maximum(xs) for xs in zip(idxs...))
+end
+
+_check_input(idx::AbstractArray{<:Integer}, arr) = checkbounds(arr, minimum(idx):maximum(idx))
```

Comment:
these check functions are not general utilities, they shouldn't stay in`src/utils.jl` but where they are needed. 
Also, they should have more specific names, like `_check_gather_input`

Reply:
fixed this

Reply:
I need this in checking dimensions in `scatter` as well! I move them back in new coming commits.